src/client/datascience/dataScienceSurveyBanner.ts

```diff
@@ -87,102 +111,235 @@ export type ShowBannerWithExpiryTime = {
     expiry?: number;
 };
 @injectable()
-export class DataScienceSurveyBanner implements IJupyterExtensionBanner {
-    public get enabled(): boolean {
-        if (this.applicationEnvironment.uiKind !== UIKind.Desktop || this.applicationEnvironment.channel !== 'stable') {
+export class DataScienceSurveyBanner implements IJupyterExtensionBanner, IExtensionSingleActivationService {
+    public isEnabled(type: BannerType): boolean {
+        switch (type) {
+            case BannerType.InsidersNotebookSurvey:
+                if (this.useVSCodeNotebookEditorApi && this.applicationEnvironment.channel === 'insiders') {
+                    return this.isEnabledInternal(type);
+                }
+                break;
+            case BannerType.ExperimentNotebookSurvey:
+                if (
+                    this.applicationEnvironment.channel === 'stable' &&
+                    this.experimentService.inExperiment(Experiments.NativeNotebook)
+                ) {
+                    return this.isEnabledInternal(type);
+                }
+                break;
+            case BannerType.DSSurvey:
+            default:
+                if (this.applicationEnvironment.channel === 'stable') {
+                    return this.isEnabledInternal(type);
+                }
+        }
+        return false;
+    }
+    private isEnabledInternal(type: BannerType): boolean {
+        if (this.applicationEnvironment.uiKind !== UIKind.Desktop) {
             return false;
         }
-        if (!this.showBannerState.value.expiry) {
+
+        if (!this.showBannerState[type].value.expiry) {
             return true;
         }
-        return this.showBannerState.value.expiry! < Date.now();
+        return this.showBannerState[type].value.expiry! < Date.now();
     }
+
     private disabledInCurrentSession: boolean = false;
-    private bannerMessage: string = localize.DataScienceSurveyBanner.bannerMessage();
     private bannerLabels: string[] = [
         localize.DataScienceSurveyBanner.bannerLabelYes(),
         localize.DataScienceSurveyBanner.bannerLabelNo()
     ];
-    private readonly showBannerState: IPersistentState<ShowBannerWithExpiryTime>;
-    private readonly surveyLink: string;
+    private readonly showBannerState: IPersistentState<ShowBannerWithExpiryTime>[] = [];
```

Comment:
Seems like this should be a map, not an array. It would reenforce that it is being set correctly.

```suggestion
private readonly showBannerState = new Map<BannerType, IPersistentState<ShowBannerWithExpiryTime>>();
```